packages/math-input/src/components/tabbar/tabbar.tsx

```diff
@@ -62,4 +71,35 @@ function Tabbar(props: Props): React.ReactElement {
     );
 }
 
+// Custom focus hook to handle arrow key navigation for the TabBar for each ArrowKeyTabItem.
+function useArrowKeyFocus(size) {
+    const [currentFocus, setCurrentFocus] = useState(0);
+
+    const handleKeyDown = useCallback(
+        (e) => {
+            if (e.keyCode === 39) {
+                // Right arrow
+                e.preventDefault();
+                setCurrentFocus(
+                    currentFocus === size - 1 ? size - 1 : currentFocus + 1,
+                );
+            } else if (e.keyCode === 37) {
+                // Left arrow
+                e.preventDefault();
+                setCurrentFocus(currentFocus === 0 ? 0 : currentFocus - 1);
+            }
+        },
+        [size, currentFocus, setCurrentFocus],
+    );
+
+    useEffect(() => {
+        document.addEventListener("keydown", handleKeyDown, false);
+        return () => {
+            document.removeEventListener("keydown", handleKeyDown, false);
+        };
+    }, [handleKeyDown]);
```

Comment:
After reading [this article](https://kentcdodds.com/blog/usememo-and-usecallback), I've been a skeptic of `useCallback` without a demonstrable need. Is there a reason this doesn't work?

```suggestion
    function handleKeyDown(e) {
        if (e.keyCode === 39) {
            // Right arrow
            e.preventDefault();
            setCurrentFocus(
                currentFocus === size - 1 ? size - 1 : currentFocus + 1,
            );
        } else if (e.keyCode === 37) {
            // Left arrow
            e.preventDefault();
            setCurrentFocus(currentFocus === 0 ? 0 : currentFocus - 1);
        }
    };

    useEffect(() => {
      document.addEventListener("keydown", handleKeyDown, false);
      return () => {
          document.removeEventListener("keydown", handleKeyDown, false);
      });
```

Reply:
Totally agree with you, it was a section of code I pulled from some examples online. I haven't used useCallback very often and was looking to remove/refactor anyway as it seemed gratuitous. Deleted! :)